### PR TITLE
Fixes: #6925 Interfaces Table - bring back the visual aids from v2.9

### LINKED ISSUE #6925: Interfaces Table - bring back the visual aids from v2.9
### NetBox version

v3.0-dev

### Feature type

Change to existing functionality

### Proposed functionality

In v2.9:

1. Virtual interfaces were distinguished by their yellow row color. 
2. Disabled interfaces were distinguished by their red row color.
3. Connected interfaces were distinguished by their green row color.    
4. The interface type was distinguished by different icons before the interface name:
- Circle for virtual interfaces
- "Hamburger" for LAG interfaces
- Wrench for OOB Management interface
- Wifi symbol for wifi links
- Two arrows for ethernet links and others

image

This is something that got lost somewhere in v2.10 or v2.11.

v2.11
image

v3.0-dev
image

- All interfaces, virtual and physical have the same row color. Only connected interfaces get a green row color.
- All interfaces, despite their type, have the same icon before the interface name.
- To find the OOB Management interface, one needs to enable the "Management Only" column.
- To find the enabled interfaces, one needs to enable the "Enabled" column.

This FR proposes to bring back the visual aids that helped get instant information by looking at the interfaces table without overwhelming it with the columns: "Management Only", "Enabled" and "Type". 
This FR does not intent to remove the option of having those columns. They are still useful to order the interfaces.

### Use case

Reduce the need of the columns "Management Only", "Enabled" and "Type" on the Interfaces table by adding some visual aids that were used on past versions (v2.9).

1. Virtual interfaces could be distinguished by their yellow row color. 
2. Disabled interfaces were distinguished by their red row color.
3. Connected interfaces were distinguished by their green row color.    
4. The interface type was distinguished by different icons before the interface name.

With this feature, one can instantly get information about the interface type and status by looking at the interfaces table.
No need to overwhelm the interfaces table with extra columns: "Management Only", "Enabled" and "Type"
These columns should still be available to be added to the interfaces tables. They are useful to order the interfaces.  

### Database changes

_No response_

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by teixemf**
### Fixes: #6925 Interfaces Table - bring back the visual aids from v2.9

Reduce the need of the columns "Management Only", "Enabled" and "Type" on the Interfaces table by adding some visual aids that were used on past versions (v2.9).

1. Virtual Interfaces are distinguished by row color "primary ".
2. Disabled Interfaces are distinguished by row color "danger".
3. Keep Connected Interfaces distinguished by row color "success".
4. Interface type distinguished by different icons before the interface name:
   - Management Only: wrench
   - LAG: reorder-horizontal
   - Virtual: circle
   - Wireless: wifi
   - Others: ethernet

### CODE REVIEW

File: netbox/dcim/tables/devices.py [hunk 1]
```diff
@@ -53,6 +53,16 @@ def get_cabletermination_row_class(record):
     return ''
 
 
+def get_interface_row_class(record):
+    if not record.enabled:
+        return 'danger'
+    elif not record.is_connectable:
```
Comment (jeremystretch) [inline netbox/dcim/tables/devices.py:59]:
```suggestion
    elif record.is_virtual:
```

This should evaluate `is_virtual` to accommodate wireless interfaces in v3.1.

File: netbox/dcim/tables/devices.py [hunk 2]
```diff
@@ -53,6 +53,16 @@ def get_cabletermination_row_class(record):
...
+        return 'primary'
+    else:
+        return get_cabletermination_row_class(record)
+    return ''
```
Comment (jeremystretch) [inline netbox/dcim/tables/devices.py:63]:
```suggestion
    return get_cabletermination_row_class(record)
```

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Looks good! Just a couple minor tweaks.

**Comment by jeremystretch** [comment]
Thanks @CironAkono!

**Comment by DanSheps** [comment]
The visual aids are back!

**Pull request merged**